data/test/macros/wml_unit_test_macros.cfg

```diff
@@ -75,3 +75,63 @@
         [/then]
     [/if]
 #enddef
+
+# have one adjacent unit attack another adjacent unit
+# each unit is given 1000 hp and their attack is made to 1x100, with a 100% chance of hitting
+#define ATTACK_AND_VALIDATE DAMAGE1
+
+#arg ID1
+alice#endarg
+
+#arg ID2
+bob#endarg
+
+#arg DAMAGE2
+{DAMAGE1}#endarg
+
+    [modify_unit]
+        [filter]
+        [/filter]
```

Comment:
Suggest filtering on `id={ID1},{ID2}`, otherwise this applies the modification to other units on the map too.

Reply:
This is intentional for the resistance_affect_allies and resistance_affect_everybody tests.

Reply:
Please would you explain further? This filter isn't for which units are affected by the abilities, it's for which units get the huge max_hp, healing and 100x1 attack that has 100% CTH.

For the tests that have Charlie and Dave fight, even with the filters they would get those bonuses when it's their turn to fight.

Reply:
Updated, looks like I misread it.